src/Xamarin.Android.Build.Tasks/Tasks/GeneratePackageManagerJava.cs

```diff
@@ -413,7 +413,7 @@ void AddEnvironment ()
 					Log
 				);
 			} else {
-				marshalMethodsAsmGen = new MarshalMethodsNativeAssemblyGenerator (uniqueAssemblyNames);
+				marshalMethodsAsmGen = new MarshalMethodsNativeAssemblyGenerator (assemblyCount, uniqueAssemblyNames);
```

Comment:
Why is `assemblyCount` != `uniqueAssemblyNames.Count`?  In what instances will `assemblyCount` be greater than `uniqueAssemblyNames.Count`?  I guess `UseAssemblyStore`? https://github.com/xamarin/xamarin-android/blob/54a3f6e0bedc9f9b204a535aaf0f664e22285119/src/Xamarin.Android.Build.Tasks/Tasks/GeneratePackageManagerJava.cs#L272-L275

Reply:
Yep, it's the assembly stores case